test/jira-tests.js

```diff
@@ -0,0 +1,442 @@
+import JiraApi from '../src/jira.js';
+
+describe('Jira API Tests', () => {
+  describe('Constructor Tests', () => {
+    it('Constructor functions properly', () => {
+      const jira = new JiraApi({
+        protocol: 'http',
+        host: 'jira.somehost.com',
+        port: '8080',
+        username: 'someusername',
+        password: 'somepassword',
+        apiVersion: '2.0',
+        base: '',
+        strictSSL: true
+      });
+
+      expect(jira.protocol).to.eql('http');
+      expect(jira.host).to.eql('jira.somehost.com');
+      expect(jira.port).to.eql('8080');
+      expect(jira.username).to.eql('someusername');
+      expect(jira.password).to.eql('somepassword');
+      expect(jira.apiVersion).to.eql('2.0');
+    });
+  });
+
+  describe('makeUri Tests', () => {
+    it('makeUri functions properly in the average case', () => {
+      const jira = new JiraApi({
+        protocol: 'http',
+        host: 'jira.somehost.com',
+        port: '8080',
+        username: 'someusername',
+        password: 'somepassword',
+        apiVersion: '2.0',
+        strictSSL: true
+      });
+
+      expect(jira.makeUri('/somePathName'))
+        .to.eql('http://jira.somehost.com:8080/rest/api/2.0/somePathName');
+    });
+  });
+
+  describe('doRequest Tests', () => {
+    it('doRequest functions properly in the average case', (done) => {
+      const dummyObject = {
+        body: JSON.stringify({
+          someKey: 'someValue'
+        })
+      };
+      function dummyRequest(requestOptions) {
+        return Promise.resolve(dummyObject);
+      }
+      const jira = new JiraApi({
+        protocol: 'http',
+        host: 'jira.somehost.com',
+        port: '8080',
+        apiVersion: '2.0',
+        strictSSL: true,
+        request: dummyRequest
+      });
+
+      jira.doRequest({})
+        .should.eventually.eql(JSON.parse(dummyObject.body))
+        .and.notify(done);
+    });
+
+    it('doRequest authenticates properly when specified', (done) => {
+      function dummyRequest(requestOptions) {
+        return Promise.resolve({
+          body: JSON.stringify(requestOptions)
+        });
+      }
+      const jira = new JiraApi({
+        protocol: 'http',
+        host: 'jira.somehost.com',
+        port: '8080',
+        username: 'someusername',
+        password: 'somepassword',
+        apiVersion: '2.0',
+        strictSSL: true,
+        request: dummyRequest
+      });
```

Comment:
You seem to be using a lot of the same default options in your tests. Have you considered using a default options variable, and override them as needed?